components/modules/MentorCard/components/index.tsx

```diff
@@ -0,0 +1,231 @@
+import "twin.macro";
+import Image from "next/image";
+import { ReactElement, useMemo, useCallback, useState } from "react";
+import {
+    Stack,
+    StackDivider,
+    Badge as ChakraBadge,
+    AspectRatio,
+    IconButton,
+    IconButtonProps,
+} from "@chakra-ui/react";
+import { motion } from "framer-motion";
+import { Star, UserRectangle, ChatTeardropText, User, Heart } from "phosphor-react";
+import ColorThief from "color-thief";
+import { css } from "@emotion/css";
+
+/* -------------------------------------------------------------------------- */
+/*                                Profile Image                               */
+/* -------------------------------------------------------------------------- */
+
+// const rgbToHex = (r: number, g: number, b: number) =>
+//     `#${[r, g, b]
+//         .map((x) => {
+//             const hex = x.toString(16);
+//             return hex.length === 1 ? "0" + hex : hex;
+//         })
+//         .join("")}`;
+
+function rgbToHsl(r: number, g: number, b: number) {
+    (r /= 255), (g /= 255), (b /= 255);
+
+    const max = Math.max(r, g, b),
+        min = Math.min(r, g, b);
+    let h = 0;
+    let s = 0;
+    const l = (max + min) / 2;
+
+    if (max == min) {
+        h = s = 0; // achromatic
+    } else {
+        const d = max - min;
+        s = l > 0.5 ? d / (2 - max - min) : d / (max + min);
+
+        switch (max) {
+            case r:
+                h = (g - b) / d + (g < b ? 6 : 0);
+                break;
+            case g:
+                h = (b - r) / d + 2;
+                break;
+            case b:
+                h = (r - g) / d + 4;
+                break;
+        }
+
+        h /= 6;
+    }
+
+    return [h * 360, s * 100, l * 100];
+}
+
+/**
+ * Conditionally render a placeholder profile image
+ *
+ */
+export function ProfileImage({ url }: { url?: string }) {
+    const [hsl, setHsl] = useState([0, 0, 0]);
+    return (
+        <AspectRatio
+            w="100%"
+            ratio={1}
+            rounded="md"
+            overflow="hidden"
+            className={css`
+                box-shadow: 0 12px 24px -5px hsla(${hsl[0]}, ${Math.min(hsl[1] + 50, 100)}%, ${hsl[2]}%, 0.25);
```

Comment:
Dynamic shadow color based on prominent color of the image

Reply:
Cool! Awesome work here.